libsolidity/formal/SMTChecker.cpp

```diff
@@ -400,9 +413,9 @@ void SMTChecker::booleanOperation(BinaryOperation const& _op)
 	{
 		// @TODO check that both of them are not constant
 		if (_op.getOperator() == Token::And)
-			m_interface->addAssertion(expr(_op) == expr(_op.leftExpression()) && expr(_op.rightExpression()));
+			m_interface->addAssertion(expr(_op) == (expr(_op.leftExpression()) && expr(_op.rightExpression())));
```

Comment:
Oh  wow, thanks for that :)

Reply:
But actually - is this based on the most recent pull request? There should be a function `defineExpression(_op, ...)`